app/scripts/controllers/pod.js

```diff
@@ -86,7 +86,115 @@ angular.module('openshiftConsole')
         });
       }
     };
+       
+    $scope.onChange = function(arg) {     
+      // Iter. over all containers, setting new to active/connected & old's isActive to false
+      _.forEach($scope.containerTerminals, function(ctr){
+        if (ctr === arg) {
+          ctr.isActive = true;
+          
+          // If not previously connected to (Prevents duplicates)
+          if (!ctr.isConnected) {
+            ctr.isConnected = true;
+            // Add this to the list of connected terminals
+            $scope.connectedTerminals.push(ctr);
+          }
+        } else {
+          ctr.isActive = false;
+        }
+      }); 
+    }
+    
+    $scope.tabSelect = function() {
```

Comment:
You need the `terminalTabWasSelected`variable, but I can't see the `tabSelect()` function being used in the view.